Add deep nesting support for .get(), .set() and .del()

morkro opened the pull request:
Here is the pull request for #33 .
I've added tests, updated the README a little and bumped the version.

rexxars commented:
`del` doesn't work as expected here:

```js
conf.set('foo.bar.baz', {awesome: true});
conf.set('foo.bar.bang', {awesome: false});

conf.del('foo.bar.baz');
console.log(this.conf.get('foo.bar.bang'));
// Expected: {awesome: false}
// Actual: undefined
```

rexxars commented:
Also, the changes done in README stripped significant whitespace (two spaces + linebreak === hard break) - so it's probably best if you put those back :-)

Otherwise, looks good!

morkro commented:
Thanks, I didn't realise I removed the breaks!

I couldn't reproduce your example though. It works for me. Might be that the result is `undefined` because you're missing `this` in your example: `conf.set('foo.bar.baz', {awesome: true});`

It should be `this.conf.set('foo.bar.baz', {awesome: true});`.

morkro commented:
Okay, the breaks are somehow removed by my IDE. I added `<br>` to make them appear again. Hope that's fine.

rexxars commented:
The following should reproduce the issue - paste it into the `del()` test:

```js
this.conf.set('foo.bar.baz', {awesome: 'icecream'});
this.conf.set('foo.bar.zoo', {awesome: 'redpanda'});

this.conf.del('foo.bar.baz');
assert.equal(this.conf.get('foo.bar.zoo.awesome'), 'redpanda');
```

sindresorhus commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "configstore",
-  "version": "1.4.0",
+  "version": "2.0.0",
```
Don't bump this.

morkro replied:
Sure, will revert that. Thought I could bump it because you were talking about a major update. Sorry for that :grimacing:

sindresorhus replied:
Yes, but I have a script for that. And I might want to do other changes before doing a release.

morkro commented:
Updated to previous feedback. 
- `.del()` now uses the `.delete()` method from `dot-prop`
- reverted the version back to `1.4.0`

Hope this PR is finally working for everyone. Sorry for the issues.

sindresorhus commented:
Looks good. Thanks :)
